lib/client-assets.php

```diff
@@ -1012,6 +1012,7 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 		'titlePlaceholder'    => apply_filters( 'enter_title_here', __( 'Add title', 'gutenberg' ), $post ),
 		'bodyPlaceholder'     => apply_filters( 'write_your_story', __( 'Write your story', 'gutenberg' ), $post ),
 		'isRTL'               => is_rtl(),
+		'autosaveInterval'    => defined( 'AUTOSAVE_INTERVAL' ) ? AUTOSAVE_INTERVAL : 60,
```

Comment:
This has the impact of changing autosaves from 10 second delay to 60 second delay. While the latter may align with the classic editor, I think it's at least worth highlighting as a significant change to saving behavior, one which may be worth discussing separately in the #core-editor meeting or a separate pull request.

Reply:
True this does change the timing, happy to discuss further if you are concerned about this change.

In terms of highlighting changes from establish behavior - I think its the other way around, running autosaves at 10 second intervals would be a significant change in established behavior. The current saves in Gutenberg are not autosaves - they are in fact draft saves that create revisions and do not work with published posts.

Reply:
Regardless, it does not appear within the scope of this pull request to change the interval.

An argument might be made if it was a regression of behavior to implement autosaves with the current interval, but if anything, it's already an improvement over the previous state of Gutenberg.

It was raised because I know in the past there's been significant consideration of the UX of autosave and its role in Gutenberg as putting less of a burden on the user for explicitly saving by having autosaving which occurs more frequently in the background (cc @jasmussen). I don't know whether 10s or 60s or some other interval is necessarily the best, but it's something that shouldn't be changed on a whim.

Reply:
Agreed. If there is a design decision around increasing intervals for autosave to every 10 seconds, lets not change that. However, this is a significant change from current core behavior that is worth highlighting/documenting. Perhaps @jasmussen can help explain or give more background about why this interval was chosen.

Don't more frequent autosaves create visual distraction in the interface by constantly showing the 'Autosaving' Saved state and disabling/enabling the publish action?  I am also concerned about the increase in server load cased by the increased save frequency in Gutenberg.

Reply:
For reference, the 10 second delay was introduced in this PR: https://github.com/WordPress/gutenberg/pull/1945, I couldn't find a decision of why the delay is 10 seconds. Here is the current core autosave.js file where the interval is set: https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/js/autosave.js#L741

Reply:
As a preface, I would at any one point in time defer to technical considerations, i.e. the both of you. Not only in the name of shipping, but in the name of other deeper aspects that I might not have a fully technical grasp on, such as limiting the amount of revisions or drafts to something that makes sense.

As such, I'll give a high level overview of how I'd love to see WordPress handling saving, long-term. The short answer is: _like how Google Docs does it_. The explicit "Save" action should ideally be completely automatic, and not even a button the user has to press. As soon as a new document is created, it is _created_ (i.e. saved). Any change you make, is _saved_, ideally a few seconds after typing. The over-arching goal is that through modern technology we can spare you having to worry about losing your work. ⌘S is almost a nervous tick for anyone who grew up using Photoshop, of fear that things would crash. But we are moving towards place where "save" isn't something you need to worry about.

I'm aware of the technical issues, both with offline support, metaboxes, revisions, and so on. The fact that we do have an explicit Save button is an affordance that acknowledges that we cannot yet do without one. But any change that moves us closer to a potential future where we can remove the save button in favor of an autosave _message_, just like Google Docs, is a good step, in my book. The less the user has to think about, the better.

Reply:
I've dropped it back to 10 seconds for now 7fd7c79f9